server/src/main/java/org/elasticsearch/action/search/SearchRequestBuilder.java

```diff
@@ -251,17 +242,6 @@ public SearchRequestBuilder seqNoAndPrimaryTerm(boolean seqNoAndPrimaryTerm) {
         return this;
     }
 
-    /**
-     * Sets the boost a specific index will receive when the query is executed against it.
-     *
-     * @param index      The index to apply the boost against
-     * @param indexBoost The boost to apply to the index
-     */
-    public SearchRequestBuilder addIndexBoost(String index, float indexBoost) {
```

Comment:
same here. I'm fine with this deletion if we are sure SearchRequestBuilder is internal API and unused by client code, otherwise maybe we should deprecate it first.